src/System.IO.Pipelines/PipeFactory.cs

```diff
@@ -45,13 +45,22 @@ public IPipeReader CreateReader(Stream stream)
             }
 
             var pipe = new Pipe(_pool);
-            ExecuteCopyToAsync(pipe, stream);
+            var ignore = ExecuteCopyToAsync(pipe, stream);
             return pipe;
         }
 
-        private Task ExecuteCopyToAsync(Pipe pipe, Stream stream)
+        private async Task ExecuteCopyToAsync(Pipe pipe, Stream stream)
         {
-            return stream.CopyToAsync(pipe);
+            try
+            {
+                await stream.CopyToAsync(pipe);
+            }
+            catch (Exception ex)
+            {
+                pipe.Writer.Complete(ex);
+                return;
+            }
+            pipe.Writer.Complete();
```

Comment:
After further thought, maybe the pipe.Writer.Complete(ex/null) calls should be moved to StreamExtensions.CopyToAsync() itself.

@davidfowl Do you agree? If so, I don't think StreamExtensions.CopyToAsync() should ever throw since you should observe any exceptions by reading from the pipe.